api/fasit_test.go

```diff
@@ -157,9 +437,9 @@ func TestResourceWithArbitraryPropertyKeys(t *testing.T) {
 		MatchParam("alias", "alias").
 		Reply(200).File("testdata/fasitResponse-arbitrary-keys.json")
 
-	resource, err := fasit.getScopedResource(ResourceRequest{"alias", "DataSource"}, "dev", "app", "zone")
+	resource, appError := fasit.getScopedResource(ResourceRequest{"alias", "DataSource"}, "dev", "app", "zone")
 
-	assert.NoError(t, err)
```

Comment:
why no error?

Reply:
We've been returning a pointer to an appError and doing our checking using the standard 'if err != nil' pattern. Turns out pointers always returns a pointer to a memory address, even though you return nil from the method. We could check this using the isNil method, which casts the content of the pointer to an interface{} before validating nil, but the best thing to do would be to return an empty appError{} rather than a pointer to nil to keep the 'err != nil' pattern consistently throughout the app.
Realising that returning a pointer to nil != nil, means we've got to rewrite quite a bit.

Reply:
oi

Reply:
sorry

Reply:
Hmm. Digging a little deeper. The code work as intended, but the problem arises when we are trying to use appError to return http status codes: 

https://golang.org/doc/faq#nil_error